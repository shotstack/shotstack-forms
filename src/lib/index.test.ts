import { describe, expect } from '@jest/globals';
import Shotstack from './index';
import defaultJsonInput from '../lib/components/Form/defaultMerge.json';
import type { IParsedEditSchema } from './ShotstackEditTemplate/types';
describe('Testing Shotstack module entry point', () => {
	it('Shotstack.render() should deploy the app inside target element', () => {
		const element = document.createElement('div');
		const shotstackFormService = new Shotstack();
		shotstackFormService.render(element);
		expect(element).toMatchSnapshot();
	});

	it("On initialization, should render the form inside the an element with id 'shotstack-form-field' if any", () => {
		const element = document.createElement('div');
		element.id = 'shotstack-form-field';
		document.body.append(element);
		new Shotstack();
		expect(element).toMatchSnapshot();
	});

	it('When passed a JSON template as a parameter, the template service should parse it correctly', () => {
		const stringifiedJson = JSON.stringify(defaultJsonInput);
		const shotstackService = new Shotstack(stringifiedJson);
		expect(shotstackService.templateService.template).toEqual(defaultJsonInput);
	});

	it("When .on(event, fn) is called, it should add a 'fn' handler for that particular 'event'", () => {
		const shotstackService = new Shotstack();
		const defaultMergeField = shotstackService.templateService.result;
		const mockChange = jest.fn();
		const find = 'Hello';
		const replace = 'World';
		const newReplace = 'New World';
		const mergeEntry = { find, replace };
		const replacedEntry = { find, replace: newReplace };
		const jsonTemplate: IParsedEditSchema = { merge: [mergeEntry] };
		const expectedResultTemplate: IParsedEditSchema = { merge: [replacedEntry] };

		shotstackService.on('change', mockChange);
		shotstackService.templateService.setTemplateSource(JSON.stringify(jsonTemplate));
		expect(mockChange).toHaveBeenCalledWith(jsonTemplate, defaultMergeField);

		shotstackService.templateService.updateResultMergeFields(replacedEntry);
		expect(mockChange).toHaveBeenCalledWith(expectedResultTemplate, jsonTemplate);
		expect(mockChange).toHaveBeenCalledTimes(2);

		const mockSubmit = jest.fn();
		shotstackService.on('submit', mockSubmit);
		shotstackService.submit();
		expect(mockSubmit).toHaveBeenCalled();
		expect(mockSubmit).toHaveBeenCalledWith(expectedResultTemplate);
	});
});

describe('Testing Shotstack methods', () => {
	it('Shotstack.display(), container element should have property display = "block" ', () => {
		const newContainer = document.createElement('div');
		const shotstackService = new Shotstack(defaultJsonInput, newContainer);
		shotstackService.display();
		expect(newContainer).toMatchSnapshot();
	});

	it('Shotstack.hide(), container element should have property display = "none"', () => {
		const newContainer = document.createElement('div');
		const shotstackService = new Shotstack(defaultJsonInput, newContainer);
		shotstackService.hide();
		expect(newContainer).toMatchSnapshot();
	});

	it('Shotstack.attach(), when passed a new container it should render cleaning previous container', () => {
		const previousContainer = document.createElement('div');
		previousContainer.id = 'previous-container';
		const newContainer = document.createElement('div');
		newContainer.id = 'new-container';
		const shotstackService = new Shotstack(defaultJsonInput, previousContainer);
		shotstackService.attach(newContainer);
		expect(previousContainer.childNodes).toHaveLength(0);
		expect(newContainer).toMatchSnapshot();
	});

	it('Shotstack.remove(), should remove content of the container where the component is rendered on', () => {
		const newContainer = document.createElement('div');
		const shotstackService = new Shotstack(defaultJsonInput, newContainer);
		shotstackService.remove();
		expect(newContainer.childNodes).toHaveLength(0);
		expect(newContainer).toMatchSnapshot();
	});
	it('Shotstack.container should return container where the component is being rendered on', () => {
		const containerElement = document.createElement('div');
		containerElement.id = 'container-element';
		const shotstackService = new Shotstack(defaultJsonInput, containerElement);
		expect(shotstackService.container).toBe(containerElement);
	});

	it('Shotstack.merge(), should return merged JSON template', () => {
		const shotstackService = new Shotstack();

		const find = 'Hello';
		const replace = 'World';
		const newReplace = 'New World';
		const mergeEntry = { find, replace };
		const replacedEntry = { find, replace: newReplace };
		const jsonTemplate: IParsedEditSchema = { merge: [mergeEntry] };
		const expectedResultTemplate: IParsedEditSchema = { merge: [replacedEntry] };

		shotstackService.templateService.setTemplateSource(JSON.stringify(jsonTemplate));
		shotstackService.templateService.updateResultMergeFields(replacedEntry);

		expect(shotstackService.merge()).toEqual(expectedResultTemplate);
	});
});
