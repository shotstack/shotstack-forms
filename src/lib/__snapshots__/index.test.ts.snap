// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Shotstack methods Shotstack.attach(), when passed a new container it should render cleaning previous container 1`] = `
<div
  id="new-container"
>
  <div
    class="shotstack-mergefield-form svelte-r93edu"
  >
    <section
      class="max-w-lg my-4 mx-auto border rounded-xl px-7 py-4 svelte-r93edu"
      data-cy="form-container"
    >
      <form
        class="svelte-r93edu"
      >
        <div
          class="mb-6 svelte-r93edu"
          data-cy="template-input-section"
        >
          <label
            class="text-teal-400 px-1 svelte-r93edu"
            for="json-input"
          >
            Paste template
          </label>
           
          <textarea
            class="w-full h-60 monospace border p-4 overflow-auto whitespace-pre resize-none svelte-r93edu"
            data-cy="template-input"
            id="json-input"
          />
        </div>
         
        <div
          class="hidden"
        >
          <p
            class="bg-rose-200 rounded py-2 my-4 px-4"
            data-cy="template-input-error"
          >
            <span
              class="monospace text-orange-900"
            >
              
            </span>
          </p>
           
          <div
            class="flex flex-row-reverse "
          >
            <button
              class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-3 rounded align-self-end"
            >
              Reset
            </button>
          </div>
        </div>
         
        <div>
          <div
            data-cy="merge-fields-input-section"
          >
            <h1
              class="text-teal-400 px-1"
            >
              Modify Merge Values
            </h1>
             
            <div
              class="border p-4 mb-6"
            >
              <div
                class="relative"
                data-cy="label-input"
              >
                <label
                  class="block mb-2 monospace"
                  for="NAME"
                >
                  NAME
                </label>
                 
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  role="textbox"
                  type="text"
                />
                 
                <button
                  class="bg-blue-400 hover:bg-blue-700 text-white font-bold absolute top-0 right-0 rounded-full w-4 h-4 p-1"
                >
                  <svg
                    viewBox="0 0 16 16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M.293.293a1 1 0 011.414 0L8 6.586 14.293.293a1 1 0 111.414 1.414L9.414 8l6.293 6.293a1 1 0 01-1.414 1.414L8 9.414l-6.293 6.293a1 1 0 01-1.414-1.414L6.586 8 .293 1.707a1 1 0 010-1.414z"
                      fill="white"
                    />
                  </svg>
                </button>
                 
              </div>
            </div>
          </div>
           
          <div>
            <div>
              <h1
                class="text-teal-400 px-1"
              >
                Add a new merge field
              </h1>
               
              <div
                class="border p-4 mb-6"
              >
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  type="text"
                />
                 
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  type="text"
                />
                 
                <div
                  class="flex flex-row-reverse"
                >
                  <button
                    class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-3 rounded align-self-end"
                  >
                    Add
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
       
      <div
        class="svelte-r93edu"
        data-cy="result-section"
      >
        <h1
          class="text-teal-400 px-1 inline-block mr-2 svelte-r93edu"
        >
          Result
        </h1>
         
        <abbr
          class="svelte-r93edu"
          title="Copy to clipboard"
        >
          <img
            alt="copy-button"
            class="h-4 cursor-pointer inline mb-1 svelte-r93edu"
            src="[object Object]"
          />
        </abbr>
         
        <p
          class="h-60 overflow-auto border p-4 whitespace-pre monospace svelte-r93edu"
          data-cy="result"
        >
          {
  "merge": [
    {
      "find": "NAME",
      "replace": "world"
    }
  ],
  "timeline": {
    "tracks": [
      {
        "clips": [
          {
            "asset": {
              "type": "title",
              "text": "Hello {{ NAME }}",
              "style": "future"
            },
            "start": 0,
            "length": 5
          }
        ]
      }
    ]
  },
  "output": {
    "format": "mp4",
    "resolution": "hd"
  }
}
        </p>
         
        <div
          class="flex justify-between pt-4"
        >
          <a
            class="bg-gray-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center w-40 justify-center"
            download="result.json"
            href=""
          >
            <svg
              class="fill-current w-4 h-4 mr-2"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 8V2H7v6H2l8 8 8-8h-5zM0 18h20v2H0v-2z"
              />
            </svg>
            
	Download
          </a>
           
          <button
            class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded w-40"
          >
            Submit
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Testing Shotstack methods Shotstack.display(), container element should have property display = "block"  1`] = `
<div
  style="display: block;"
>
  <div
    class="shotstack-mergefield-form svelte-r93edu"
  >
    <section
      class="max-w-lg my-4 mx-auto border rounded-xl px-7 py-4 svelte-r93edu"
      data-cy="form-container"
    >
      <form
        class="svelte-r93edu"
      >
        <div
          class="mb-6 svelte-r93edu"
          data-cy="template-input-section"
        >
          <label
            class="text-teal-400 px-1 svelte-r93edu"
            for="json-input"
          >
            Paste template
          </label>
           
          <textarea
            class="w-full h-60 monospace border p-4 overflow-auto whitespace-pre resize-none svelte-r93edu"
            data-cy="template-input"
            id="json-input"
          />
        </div>
         
        <div
          class="hidden"
        >
          <p
            class="bg-rose-200 rounded py-2 my-4 px-4"
            data-cy="template-input-error"
          >
            <span
              class="monospace text-orange-900"
            >
              
            </span>
          </p>
           
          <div
            class="flex flex-row-reverse "
          >
            <button
              class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-3 rounded align-self-end"
            >
              Reset
            </button>
          </div>
        </div>
         
        <div>
          <div
            data-cy="merge-fields-input-section"
          >
            <h1
              class="text-teal-400 px-1"
            >
              Modify Merge Values
            </h1>
             
            <div
              class="border p-4 mb-6"
            >
              <div
                class="relative"
                data-cy="label-input"
              >
                <label
                  class="block mb-2 monospace"
                  for="NAME"
                >
                  NAME
                </label>
                 
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  role="textbox"
                  type="text"
                />
                 
                <button
                  class="bg-blue-400 hover:bg-blue-700 text-white font-bold absolute top-0 right-0 rounded-full w-4 h-4 p-1"
                >
                  <svg
                    viewBox="0 0 16 16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M.293.293a1 1 0 011.414 0L8 6.586 14.293.293a1 1 0 111.414 1.414L9.414 8l6.293 6.293a1 1 0 01-1.414 1.414L8 9.414l-6.293 6.293a1 1 0 01-1.414-1.414L6.586 8 .293 1.707a1 1 0 010-1.414z"
                      fill="white"
                    />
                  </svg>
                </button>
                 
              </div>
            </div>
          </div>
           
          <div>
            <div>
              <h1
                class="text-teal-400 px-1"
              >
                Add a new merge field
              </h1>
               
              <div
                class="border p-4 mb-6"
              >
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  type="text"
                />
                 
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  type="text"
                />
                 
                <div
                  class="flex flex-row-reverse"
                >
                  <button
                    class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-3 rounded align-self-end"
                  >
                    Add
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
       
      <div
        class="svelte-r93edu"
        data-cy="result-section"
      >
        <h1
          class="text-teal-400 px-1 inline-block mr-2 svelte-r93edu"
        >
          Result
        </h1>
         
        <abbr
          class="svelte-r93edu"
          title="Copy to clipboard"
        >
          <img
            alt="copy-button"
            class="h-4 cursor-pointer inline mb-1 svelte-r93edu"
            src="[object Object]"
          />
        </abbr>
         
        <p
          class="h-60 overflow-auto border p-4 whitespace-pre monospace svelte-r93edu"
          data-cy="result"
        >
          {
  "merge": [
    {
      "find": "NAME",
      "replace": "world"
    }
  ],
  "timeline": {
    "tracks": [
      {
        "clips": [
          {
            "asset": {
              "type": "title",
              "text": "Hello {{ NAME }}",
              "style": "future"
            },
            "start": 0,
            "length": 5
          }
        ]
      }
    ]
  },
  "output": {
    "format": "mp4",
    "resolution": "hd"
  }
}
        </p>
         
        <div
          class="flex justify-between pt-4"
        >
          <a
            class="bg-gray-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center w-40 justify-center"
            download="result.json"
            href=""
          >
            <svg
              class="fill-current w-4 h-4 mr-2"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 8V2H7v6H2l8 8 8-8h-5zM0 18h20v2H0v-2z"
              />
            </svg>
            
	Download
          </a>
           
          <button
            class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded w-40"
          >
            Submit
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Testing Shotstack methods Shotstack.hide(), container element should have property display = "none" 1`] = `
<div
  style="display: none;"
>
  <div
    class="shotstack-mergefield-form svelte-r93edu"
  >
    <section
      class="max-w-lg my-4 mx-auto border rounded-xl px-7 py-4 svelte-r93edu"
      data-cy="form-container"
    >
      <form
        class="svelte-r93edu"
      >
        <div
          class="mb-6 svelte-r93edu"
          data-cy="template-input-section"
        >
          <label
            class="text-teal-400 px-1 svelte-r93edu"
            for="json-input"
          >
            Paste template
          </label>
           
          <textarea
            class="w-full h-60 monospace border p-4 overflow-auto whitespace-pre resize-none svelte-r93edu"
            data-cy="template-input"
            id="json-input"
          />
        </div>
         
        <div
          class="hidden"
        >
          <p
            class="bg-rose-200 rounded py-2 my-4 px-4"
            data-cy="template-input-error"
          >
            <span
              class="monospace text-orange-900"
            >
              
            </span>
          </p>
           
          <div
            class="flex flex-row-reverse "
          >
            <button
              class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-3 rounded align-self-end"
            >
              Reset
            </button>
          </div>
        </div>
         
        <div>
          <div
            data-cy="merge-fields-input-section"
          >
            <h1
              class="text-teal-400 px-1"
            >
              Modify Merge Values
            </h1>
             
            <div
              class="border p-4 mb-6"
            >
              <div
                class="relative"
                data-cy="label-input"
              >
                <label
                  class="block mb-2 monospace"
                  for="NAME"
                >
                  NAME
                </label>
                 
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  role="textbox"
                  type="text"
                />
                 
                <button
                  class="bg-blue-400 hover:bg-blue-700 text-white font-bold absolute top-0 right-0 rounded-full w-4 h-4 p-1"
                >
                  <svg
                    viewBox="0 0 16 16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M.293.293a1 1 0 011.414 0L8 6.586 14.293.293a1 1 0 111.414 1.414L9.414 8l6.293 6.293a1 1 0 01-1.414 1.414L8 9.414l-6.293 6.293a1 1 0 01-1.414-1.414L6.586 8 .293 1.707a1 1 0 010-1.414z"
                      fill="white"
                    />
                  </svg>
                </button>
                 
              </div>
            </div>
          </div>
           
          <div>
            <div>
              <h1
                class="text-teal-400 px-1"
              >
                Add a new merge field
              </h1>
               
              <div
                class="border p-4 mb-6"
              >
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  type="text"
                />
                 
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  type="text"
                />
                 
                <div
                  class="flex flex-row-reverse"
                >
                  <button
                    class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-3 rounded align-self-end"
                  >
                    Add
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
       
      <div
        class="svelte-r93edu"
        data-cy="result-section"
      >
        <h1
          class="text-teal-400 px-1 inline-block mr-2 svelte-r93edu"
        >
          Result
        </h1>
         
        <abbr
          class="svelte-r93edu"
          title="Copy to clipboard"
        >
          <img
            alt="copy-button"
            class="h-4 cursor-pointer inline mb-1 svelte-r93edu"
            src="[object Object]"
          />
        </abbr>
         
        <p
          class="h-60 overflow-auto border p-4 whitespace-pre monospace svelte-r93edu"
          data-cy="result"
        >
          {
  "merge": [
    {
      "find": "NAME",
      "replace": "world"
    }
  ],
  "timeline": {
    "tracks": [
      {
        "clips": [
          {
            "asset": {
              "type": "title",
              "text": "Hello {{ NAME }}",
              "style": "future"
            },
            "start": 0,
            "length": 5
          }
        ]
      }
    ]
  },
  "output": {
    "format": "mp4",
    "resolution": "hd"
  }
}
        </p>
         
        <div
          class="flex justify-between pt-4"
        >
          <a
            class="bg-gray-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center w-40 justify-center"
            download="result.json"
            href=""
          >
            <svg
              class="fill-current w-4 h-4 mr-2"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 8V2H7v6H2l8 8 8-8h-5zM0 18h20v2H0v-2z"
              />
            </svg>
            
	Download
          </a>
           
          <button
            class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded w-40"
          >
            Submit
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Testing Shotstack methods Shotstack.remove(), should remove content of the container where the component is rendered on 1`] = `<div />`;

exports[`Testing Shotstack module entry point On initialization, should render the form inside the an element with id 'shotstack-form-field' if any 1`] = `
<div
  id="shotstack-form-field"
>
  <div
    class="shotstack-mergefield-form svelte-r93edu"
  >
    <section
      class="max-w-lg my-4 mx-auto border rounded-xl px-7 py-4 svelte-r93edu"
      data-cy="form-container"
    >
      <form
        class="svelte-r93edu"
      >
        <div
          class="mb-6 svelte-r93edu"
          data-cy="template-input-section"
        >
          <label
            class="text-teal-400 px-1 svelte-r93edu"
            for="json-input"
          >
            Paste template
          </label>
           
          <textarea
            class="w-full h-60 monospace border p-4 overflow-auto whitespace-pre resize-none svelte-r93edu"
            data-cy="template-input"
            id="json-input"
          />
        </div>
         
        <div
          class="hidden"
        >
          <p
            class="bg-rose-200 rounded py-2 my-4 px-4"
            data-cy="template-input-error"
          >
            <span
              class="monospace text-orange-900"
            >
              
            </span>
          </p>
           
          <div
            class="flex flex-row-reverse "
          >
            <button
              class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-3 rounded align-self-end"
            >
              Reset
            </button>
          </div>
        </div>
         
        <div>
          <div
            data-cy="merge-fields-input-section"
          >
            <h1
              class="text-teal-400 px-1"
            >
              Modify Merge Values
            </h1>
             
            <div
              class="border p-4 mb-6"
            />
          </div>
           
          <div>
            <div>
              <h1
                class="text-teal-400 px-1"
              >
                Add a new merge field
              </h1>
               
              <div
                class="border p-4 mb-6"
              >
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  type="text"
                />
                 
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  type="text"
                />
                 
                <div
                  class="flex flex-row-reverse"
                >
                  <button
                    class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-3 rounded align-self-end"
                  >
                    Add
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
       
      <div
        class="svelte-r93edu"
        data-cy="result-section"
      >
        <h1
          class="text-teal-400 px-1 inline-block mr-2 svelte-r93edu"
        >
          Result
        </h1>
         
        <abbr
          class="svelte-r93edu"
          title="Copy to clipboard"
        >
          <img
            alt="copy-button"
            class="h-4 cursor-pointer inline mb-1 svelte-r93edu"
            src="[object Object]"
          />
        </abbr>
         
        <p
          class="h-60 overflow-auto border p-4 whitespace-pre monospace svelte-r93edu"
          data-cy="result"
        >
          {
  "merge": []
}
        </p>
         
        <div
          class="flex justify-between pt-4"
        >
          <a
            class="bg-gray-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center w-40 justify-center"
            download="result.json"
            href=""
          >
            <svg
              class="fill-current w-4 h-4 mr-2"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 8V2H7v6H2l8 8 8-8h-5zM0 18h20v2H0v-2z"
              />
            </svg>
            
	Download
          </a>
           
          <button
            class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded w-40"
          >
            Submit
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Testing Shotstack module entry point Shotstack.render() should deploy the app inside target element 1`] = `
<div>
  <div
    class="shotstack-mergefield-form svelte-r93edu"
  >
    <section
      class="max-w-lg my-4 mx-auto border rounded-xl px-7 py-4 svelte-r93edu"
      data-cy="form-container"
    >
      <form
        class="svelte-r93edu"
      >
        <div
          class="mb-6 svelte-r93edu"
          data-cy="template-input-section"
        >
          <label
            class="text-teal-400 px-1 svelte-r93edu"
            for="json-input"
          >
            Paste template
          </label>
           
          <textarea
            class="w-full h-60 monospace border p-4 overflow-auto whitespace-pre resize-none svelte-r93edu"
            data-cy="template-input"
            id="json-input"
          />
        </div>
         
        <div
          class="hidden"
        >
          <p
            class="bg-rose-200 rounded py-2 my-4 px-4"
            data-cy="template-input-error"
          >
            <span
              class="monospace text-orange-900"
            >
              
            </span>
          </p>
           
          <div
            class="flex flex-row-reverse "
          >
            <button
              class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-3 rounded align-self-end"
            >
              Reset
            </button>
          </div>
        </div>
         
        <div>
          <div
            data-cy="merge-fields-input-section"
          >
            <h1
              class="text-teal-400 px-1"
            >
              Modify Merge Values
            </h1>
             
            <div
              class="border p-4 mb-6"
            />
          </div>
           
          <div>
            <div>
              <h1
                class="text-teal-400 px-1"
              >
                Add a new merge field
              </h1>
               
              <div
                class="border p-4 mb-6"
              >
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  type="text"
                />
                 
                <input
                  class="border w-full mb-3 pl-2 py-1 text-stone-500"
                  type="text"
                />
                 
                <div
                  class="flex flex-row-reverse"
                >
                  <button
                    class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-3 rounded align-self-end"
                  >
                    Add
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
       
      <div
        class="svelte-r93edu"
        data-cy="result-section"
      >
        <h1
          class="text-teal-400 px-1 inline-block mr-2 svelte-r93edu"
        >
          Result
        </h1>
         
        <abbr
          class="svelte-r93edu"
          title="Copy to clipboard"
        >
          <img
            alt="copy-button"
            class="h-4 cursor-pointer inline mb-1 svelte-r93edu"
            src="[object Object]"
          />
        </abbr>
         
        <p
          class="h-60 overflow-auto border p-4 whitespace-pre monospace svelte-r93edu"
          data-cy="result"
        >
          {
  "merge": []
}
        </p>
         
        <div
          class="flex justify-between pt-4"
        >
          <a
            class="bg-gray-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center w-40 justify-center"
            download="result.json"
            href=""
          >
            <svg
              class="fill-current w-4 h-4 mr-2"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 8V2H7v6H2l8 8 8-8h-5zM0 18h20v2H0v-2z"
              />
            </svg>
            
	Download
          </a>
           
          <button
            class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded w-40"
          >
            Submit
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;
